### PR TITLE
chore: upgrade web.py dependency

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

* upgrade web3.py dependency to 5.27.0
* this required also updating eth-account to 0.5.7

### How to verify it

should maybe check if docker image builds first

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

**Comment by unparalleled-js** [comment]
Running into issue with this. Standby!

**Review by fubuloubu — APPROVED** [review]

**Review by evanblank3 — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -94,7 +94,7 @@
         "rich>=10.14,<11",
         "tqdm>=4.62.3,<5.0",
         "typing-extensions ; python_version<'3.8'",
-        "web3[tester]==5.27.0",
+        "web3[tester]==5.26.0",
```
Comment (fubuloubu) [inline setup.py:97]: I guess this doesn't work after all?

**Review by fubuloubu — COMMENTED** [review]
Maybe should just close this for now until all things are working that we can loosen the pin a bit

File: setup.py (hunk 1 above)
Comment (unparalleled-js) [reply setup.py:97]: Oh oops I don't think I meant to undo that.
I had some weird issue where I had accidentally pushed this to my `origin/main`. I must have mistakenly reset this here.

**Comment by unparalleled-js** [comment]
> Maybe should just close this for now until all things are working that we can loosen the pin a bit

From what I gather, this is the last release we can be on before having to figure out how to install `typing-extensions>=4.0`.
We need this version for `ape-hardhat` to work and be able to remove the pending-block hack

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**